cogs/images.py

```diff
@@ -20,45 +20,80 @@
 from discord.ext import commands
 
 # misc imports
+import concurrent.futures
 import os
 import numpy as np
 import cv2
 import math
-from functools import wraps
+from functools import wraps, partial
 from io import BytesIO
 
+MAX_IMAGE_SIZE = 8 * (10**6)
+
 
 def filter_image(func):
     @wraps(func)
     async def wrapper(self, ctx, *args):
         att = await Images.get_attachment(ctx)
         if att is None:
-            await ctx.send(
-                "no image could be found (only attached image files"
-                " can be detected) or message could not be found",
-                delete_after=15)
+            await ctx.send("no image could be found (only attached image files"
+                           " can be detected) or message could not be found")
             return
 
         await ctx.trigger_typing()
-
         buffer = BytesIO()
         await att.save(fp=buffer)
+
         original_name, ext = att.filename.rsplit('.', 1)
+        ext = ext.lower()
+        if ext in ("jpeg", "jpg"):
+            is_png = False
+        elif ext in ("png"):
+            is_png = True
+        else:
+            await ctx.send("image format not supported.")
+            return
+
         fn = f"{original_name}-{func.__name__}.{ext}"
+        ratio = (MAX_IMAGE_SIZE /
+                 att.size) * 100 if att.size > MAX_IMAGE_SIZE else 100
 
         try:
-            img_bytes = np.asarray(bytearray(buffer.read()), dtype=np.uint8)
```

Comment:
maybe better to throw this whole chunk of code into a single lambda and give it to the executor instead of translating line by line

Reply:
lambdas can't be passed through `run_in_executor` due to a pickling issue i don't fully understand

Reply:
refactored to do this